src/components/Modal/SearchModal.js

```diff
@@ -1,19 +1,25 @@
-import React, { useRef, useState, useEffect } from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import { useHistory } from 'react-router-dom';
 import styled from 'styled-components';
+import { API } from '../../config';
 import SearchListModal from './SearchListModal';
 import TagModal from './TagModal';
 
 function SearchModal({ clickSearch }) {
-  const [text, setText] = useState('');
-  const history = useHistory();
   const [postingInfo, setPostingInfo] = useState([]);
   const [searchText, setSearchText] = useState('');
+  const timerRef = useRef(-1);
```

Comment:
초기값 그냥 undefined로 줘도 될 것 같습니다!
```suggestion
  const timerRef = useRef()
  
  if(timerRef.current){
  ~~~~
  };
```